scripts/simple.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <stdint.h>
+
+struct script_context {
```

Comment:
Might be worth putting this into a header that can be easily included in a plugin?

Would we need any versioning of the interface?

Reply:
Yes, we can think about it. I've just put the struct as it looks very simple to copy and paste.

My concern is about having `struct list` for argument info because it adds a dependency on the internal list implementation. It's not always required so user can decide not to use and remove the dependency. I will think more about it. Thanks for the comment.

Reply:
> looks very simple to copy and paste.

Depends if we're ever going to change the structure. I guess as long as we put new members at the end we will be fine.

Depends if you think it would be useful for plugins to know the argument types? Can always follow up with that later.

In the future we may want to have a `uftrace/plugin.h` (or similar) that includes all the necessary headers for parsing the argument and this structure. For now, it's likely fine to ignore the arguments.

Reply:
Thanks. `uftrace/plugin.h` looks good. We have to decide how to provide the header to users. I also prototyped another feature that needs to provide a header to user, but it's also pending now at https://github.com/namhyung/uftrace/pull/636.

I will think more about how to provide those headers.

Reply:
I agree that we should provide a header for API/ABI compatibility and also it's good to use a versioning.  Then we should consider installing/loading the header and plugin binaries as well.

Reply:
I've made a draft change at c2c22b8c regarding versioning and providing a header `uftrace-script.h`.

The C/C++ scripts can be written by including `uftrace-script.h` and be compiled. The version can be checked before using the struct. We can change the major version when the layout is change and the minor version can be changed when we added a new members at the end of struct.

It's just a draft version, but would like to hear how you think about this approach.